samples/dnn/person_reid.cpp

```diff
@@ -1,245 +1,420 @@
     {
         ret.push_back(feature[i] / sum);
     }
     return ret;
 }
 
-static void extractFeatures(const std::vector<std::string>& imglist, Net* net, const int& batch_size, const int& resize_h, const int& resize_w, std::vector<std::vector<float>>& features)
+static void extractFeatures(vector<Mat> &imglist, Net *net, vector<vector<float>> &features)
 {
-    for(int st = 0; st < (int)imglist.size(); st += batch_size)
+    for (int st = 0; st < (int)imglist.size(); st++)
     {
-        std::vector<Mat> batch;
-        for(int delta = 0; delta < batch_size && st + delta < (int)imglist.size(); delta++)
-        {
-            Mat img = imread(imglist[st + delta]);
-            batch.push_back(preprocess(img));
-        }
-        Mat blob = dnn::blobFromImages(batch, 1.0, Size(resize_w, resize_h), Scalar(0.0,0.0,0.0), true, false, CV_32F);
+        Mat processed_img = preprocess(imglist[st]);
+
+        Mat blob = blobFromImage(processed_img, 1.0, Size(resize_w, resize_h), Scalar(), true, false, CV_32F);
         net->setInput(blob);
-        Mat out = net->forward();
-        for(int i = 0; i < (int)out.size().height; i++)
+        Mat out=net->forward();
+        vector<int> s {out.size[0], out.size[1]};
+        out = out.reshape(1, s);
+        for (int i = 0; i < out.rows; i++)
         {
-            std::vector<float> temp_feature;
-            for(int j = 0; j < (int)out.size().width; j++)
+            vector<float> temp_feature;
+            for (int j = 0; j < out.cols; j++)
             {
                 temp_feature.push_back(out.at<float>(i,j));
             }
             features.push_back(normalization(temp_feature));
         }
     }
-    return ;
+    return;
 }
 
-static void getNames(const std::string& ImageList, std::vector<std::string>& result)
-{
-    std::ifstream img_in(ImageList);
-    std::string img_name;
-    while(img_in >> img_name)
-    {
-        result.push_back(img_name);
-    }
-    return ;
-}
-
-static float similarity(const std::vector<float>& feature1, const std::vector<float>& feature2)
+static float similarity(const vector<float> &feature1, const vector<float> &feature2)
 {
     float result = 0.0;
-    for(int i = 0; i < (int)feature1.size(); i++)
+    for (int i = 0; i < (int)feature1.size(); i++)
     {
         result += feature1[i] * feature2[i];
     }
     return result;
 }
 
-static void getTopK(const std::vector<std::vector<float>>& queryFeatures, const std::vector<std::vector<float>>& galleryFeatures, const int& topk, std::vector<std::vector<int>>& result)
+static int getTopK(const vector<vector<float>> &queryFeatures, const vector<vector<float>> &galleryFeatures)
 {
-    for(int i = 0; i < (int)queryFeatures.size(); i++)
+    if (queryFeatures.empty() || galleryFeatures.empty())
+        return -1; // No valid index if either feature list is empty
+
+    int bestIndex = -1;
+    float maxSimilarity = -1000000000.0;
+
+    const vector<float> &query = queryFeatures[0];
```

Comment:
The first element is used. I propose to replace the function parameter to `const std::vector<float>&` and check query emptiness on caller side.